sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -733,4 +761,45 @@ private DisplayData build() {
 
     return Item.create(key, type, value);
   }
+
+  /**
+   * Wraps exceptions passed via {@link #from(Throwable)}.
+   *
+   * @see #from(Throwable)
+   */
+  static class DisplayDataException extends Exception implements HasDisplayData {
```

Comment:
I don't know if you want to create a wrapper exception like this. Why not just take the exceptino passed in, and use that to make a DisplayData?